### ISSUE TITLE
Suggest `pin::pin!` over `Box::pin` in diagnostics

### ISSUE THREAD

**Issue by yoshuawuyts** [rust-lang/rust#109964]
### Code

Consider the following code, taken from the [async book](https://rust-lang.github.io/async-book/04_pinning/01_chapter.html) ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d9805a3c1015e5283496e6bd8582a9f8)):

```Rust
pub fn main() {
    let mut test1 = Test::new("test1");
    let mut test1 = unsafe { Pin::new_unchecked(&mut test1) };
    Test::init(test1.as_mut());

    let mut test2 = Test::new("test2");
    let mut test2 = unsafe { Pin::new_unchecked(&mut test2) };
    Test::init(test2.as_mut());

    println!("a: {}, b: {}", Test::a(test1.as_ref()), Test::b(test1.as_ref()));
    std::mem::swap(test1.get_mut(), test2.get_mut());
    println!("a: {}, b: {}", Test::a(test2.as_ref()), Test::b(test2.as_ref()));
}
```

### Current output

```Shell
error[E0277]: `PhantomPinned` cannot be unpinned
  --> src/main.rs:47:26
   |
47 |     std::mem::swap(test1.get_mut(), test2.get_mut());
   |                          ^^^^^^^ within `Test`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using `Box::pin`
note: required because it appears within the type `Test`
```

### Desired output

```Shell
error[E0277]: `PhantomPinned` cannot be unpinned
  --> src/main.rs:47:26
   |
47 |     std::mem::swap(test1.get_mut(), test2.get_mut());
   |                          ^^^^^^^ within `Test`, the trait `Unpin` is not implemented for `PhantomPinned`
   |
   = note: consider using `core::pin::pin!`
note: required because it appears within the type `Test`
```

### Rationale and extra context

Local pin construction was stabilized in [Rust 1.68](https://blog.rust-lang.org/2023/03/09/Rust-1.68.0.html). We should update the diagnostics to suggest this instead of the more expensive `Box::pin` call.

### Anything else?

cc/ @wg-async

This is probably a "good first issue" as well. If someone can tag it as such?

**Comment by ndrewxie** [comment]
Alright, the suggestion has been added in #109988. IDK how to link a PR to an issue though, should I just reference the issue in the description of the PR?

**Comment by jyn514** [comment]
> IDK how to link a PR to an issue though, should I just reference the issue in the description of the PR

if you put "fixes #109964" in the PR description, it will automatically close this issue when the PR is merged :)
